File: src/Classic/VB_PerMonitorAware/MainForm.vb
Imports System.Collections.Specialized
Imports System.Configuration

Public Class MainForm

    Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.

    End Sub
    Private Sub showLayoutButton_Click(ByVal sender As Object, ByVal e As EventArgs) Handles showLayoutButton.Click
        Call New SimpleLayout().Show()
    End Sub

    ' Install targeting pack from the appropriate build
    ' \\vsufile\patches\sign\NETFX\4.7\S112.2\02032.00\MTPack\NDP463-TargetingPack-KB3183844.exe

    Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
        Dim settingsCollection As NameValueCollection = Nothing
        Try
            settingsCollection = TryCast(ConfigurationManager.GetSection("System.Windows.Forms.ApplicationConfigurationSection"), NameValueCollection)
        Catch
        End Try

        If settingsCollection IsNot Nothing Then
            dpiAwarenessLabel.Text = $"Application DPI Awareness is {settingsCollection.Get("DpiAwareness")}."
        End If
        formSizeLabel.Text = $"Form size is {Size.Width.ToString()} x {Size.Height.ToString()}"
    End Sub

    Private Sub MainForm_SizeChanged(ByVal sender As Object, ByVal e As EventArgs) Handles Me.SizeChanged
        formSizeLabel.Text = $"Form size is {Size.Width.ToString()} x {Size.Height.ToString()}"
    End Sub

    Private Sub formSizeLabel_DpiChangedAfterParent(ByVal sender As Object, ByVal e As EventArgs) Handles formSizeLabel.DpiChangedAfterParent
        formSizeLabel.Text = $"Form size is {Size.Width.ToString()} x {Size.Height.ToString()}"
    End Sub

    Private Sub customDrawing_Click(ByVal sender As Object, ByVal e As EventArgs) Handles customDrawing.Click
        CType(New CustomDrawing(), CustomDrawing).Show()
    End Sub

    'Private Sub buttonImages_Click(ByVal sender As Object, ByVal e As EventArgs)
    '    CType(New ScaleButtonImages(), ScaleButtonImages).Show()
    'End Sub

End Class

File: src/Core/CS_PerMonitorAware/MainForm.cs
﻿using Newtonsoft.Json;
using System;
using System.Collections.Specialized;
using System.Configuration;
using System.Drawing;
using System.IO;
using System.Windows.Forms;

namespace PerMonitorAware
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
            this.Icon = new Icon(new MemoryStream(JsonConvert.
                DeserializeObject<byte[]>(File.ReadAllText(".\\Resources\\wfc.json"))));
        }

        private void showLayoutButton_Click(object sender, EventArgs e)
        {
            (new SimpleLayout()).Show();
        }

        // Install targeting pack from the appropriate build
        // \\vsufile\patches\sign\NETFX\4.7\S112.2\02032.00\MTPack\NDP463-TargetingPack-KB3183844.exe

        private void MainForm_Load(object sender, EventArgs e)
        {
            NameValueCollection settingsCollection = null;
            try
            {
                settingsCollection = ConfigurationManager.GetSection("System.Windows.Forms.ApplicationConfigurationSection") as NameValueCollection;
            }
            catch
            {
            }

            //dpiAwarenessLabel.Text = $"Application DPI Awareness is {Application.HighDpiMode.ToString()}.";
            formSizeLabel.Text = $"Form size is {Size.Width.ToString()} x {Size.Height.ToString()}";
        }

        private void MainForm_SizeChanged(object sender, EventArgs e)
        {
            formSizeLabel.Text = $"Form size is {Size.Width.ToString()} x {Size.Height.ToString()}";
        }

        private void formSizeTextBox_DpiChangedAfterParent(object sender, EventArgs e)
        {
            formSizeLabel.Text = $"Form size is {Size.Width.ToString()} x {Size.Height.ToString()}";
        }

        private void customDrawing_Click(object sender, EventArgs e)
        {
            (new CustomDrawing()).Show();
        }

        private void buttonImages_Click(object sender, EventArgs e)
        {
            (new ScaleButtonImages()).Show();
        }
    }
}
